cmd/cloudshell_open/docker.go

```diff
@@ -28,11 +33,28 @@ func build(dir, image string) error {
 	return nil
 }
 
-func push(image string) error {
+func dockerPush(image string) (err error) {
+	fmt.Println(infoPrefix + " FYI, running the following command:")
+	cmdColor.Printf("\tdocker push %s\n", parameter(image))
+	end := logProgress("Pushing container image...",
```

Comment:
Lets not move printing code in here. It gets really spaghetti. 

Instead we can say “Building & pushing” in main().

Reply:
Feels like we should move the logic in build() to main()

This way we can keep logging stuff there too

Reply:
I think the main should be a little thinner TBH. Its near impossible to test as is.

Reply:
I think logging in most Go projects is buck shot throughout the project by passing a "logger" of some sort (e.g., `log.Logger`).

In this case, as everything is in the main package and utilizing globals, I think the real problem is that the different functions need to be extracted. I figured that was outside the scope of this PR.

Reply:
don’t worry about testing it, as long as we keep the logic there trivial. 

It would be nice if the path we choose for building is trivial and doesn’t require testing. My main concern is moving the UI code out of main() where it’s less discoverable.

Reply:
I inlined most of this along with the logging.